bpf/lib/nodeport.h

```diff
@@ -2078,6 +2078,47 @@ nodeport_extract_dsr_v4(struct __ctx_buff *ctx,
 	return 0;
 }
 
+/* remove_dsr_ip_opt_v4 overwrites the DSR IP Option with NOP and EOO IPv4 Options.
+ */
+static __always_inline int
+remove_dsr_ip_opt_v4(struct __ctx_buff *ctx, struct iphdr *ip4) {
+	struct dsr_opt_v4 opt;
+    __u32 iph_old, iph_new, sum_l3;
+
+	/* Check for DSR IP Option */
+	if (ip4->ihl < 0x7)
+		return CTX_ACT_OK;
+	if (ctx_load_bytes(ctx, ETH_HLEN + sizeof(struct iphdr), &opt, sizeof(opt)) < 0)
+		return DROP_INVALID;
+	if (opt.type != DSR_IPV4_OPT_TYPE)
+		return CTX_ACT_OK;
+
+	iph_old = *(__u32 *)ip4;
+
+	// Pad the DSR IP Option fields with NOPs. End with EOO.
+	// See: https://www.rfc-editor.org/rfc/rfc791.html
+	opt.type = NOP_IPV4_OPT;
+	opt.len = NOP_IPV4_OPT;
+	opt.port = 0x0101;
+	opt.addr = 0x01010100;
```

Comment:
Could this be just a `memset(&opt, IPOPT_NOOP, length);` + inserting the `IPOPT_END` as needed?

Reply:
+1, Done.